syntax = "proto3";

package rpc_bench.bw_app;

service BwService {
  rpc PushData (PbBwMessage) returns (PbBwAck) {}
  rpc StopServer (StopRequest) returns (StopResponse) {}
}

// The common header of bandwidth messages and acknowledges
message PbBwHeader {
  // randomly put some header here
  int32 field1 = 1;
  bool field2 = 2;
  string field3 = 3;
}

// The request message containing the the BwHeader and payload data
message PbBwMessage {
  // The header
  PbBwHeader header = 1;
  // A piece of unserialized (intentionally) data whose size could be very large
  bytes data = 2;
}

// The response of BwService which only contains a copy of the request's header
message PbBwAck {
  PbBwHeader header = 1;
}

message StopRequest {
  // empty
}

message StopResponse {
  // empty
}
